provider/filestore.go

```diff
@@ -26,62 +31,32 @@ import (
 	"golang.org/x/oauth2/google"
 )
 
-const (
-	Memory     pc.FileStoreType = "MEMORY"
-	FileSystem pc.FileStoreType = "LOCAL_FILESYSTEM"
-	Azure      pc.FileStoreType = "AZURE"
-	S3         pc.FileStoreType = "S3"
-	GCS        pc.FileStoreType = "GCS"
-	HDFS       pc.FileStoreType = "HDFS"
-)
-
-type FileType string
-
-const (
-	Parquet FileType = "parquet"
-	CSV     FileType = "csv"
-	DB      FileType = "db"
-)
-
-func (ft FileType) Matches(file string) bool {
-	ext := GetFileExtension(file)
-	return FileType(ext) == ft
+type FileStore interface {
+	Write(key filestore.Filepath, data []byte) error
+	Read(key filestore.Filepath) ([]byte, error)
+	Serve(keys []filestore.Filepath) (Iterator, error)
+	Exists(key filestore.Filepath) (bool, error)
+	Delete(key filestore.Filepath) error
+	DeleteAll(dir filestore.Filepath) error
+	NewestFileOfType(prefix filestore.Filepath, fileType filestore.FileType) (filestore.Filepath, error)
+	List(dirPath filestore.Filepath, fileType filestore.FileType) ([]filestore.Filepath, error)
+	NumRows(key filestore.Filepath) (int64, error)
+	Close() error
+	Upload(sourcePath filestore.Filepath, destPath filestore.Filepath) error
+	Download(sourcePath filestore.Filepath, destPath filestore.Filepath) error
+	FilestoreType() filestore.FileStoreType
+	AddEnvVars(envVars map[string]string) map[string]string
+	// CreateFilePath creates a new filepath object with the bucket and scheme from a Key
+	CreateFilePath(key string) (filestore.Filepath, error)
+	CreateDirPath(key string) (filestore.Filepath, error)
```

Comment:
Can you even create directories? You can't in S3 for example When would we create directories?